### PR TITLE
Fix: Ticketvalidation - missing cancelation popup on screen (EXPOSUREAPP-10914)

### PR DESCRIPTION
**Pull request by jurajkusnier**
### Testing
- dialog should appear only when at least one valid certificate is in the list
device-2021-12-13-112322

### Jira Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10914

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionViewModel.kt [hunk 1]
```diff
@@ -72,7 +74,11 @@ class DccTicketingCertificateSelectionViewModel @AssistedInject constructor(
         }
 
     fun onUserCancel() {
-        events.postValue(ShowCancelConfirmationDialog)
+        if (hasValidCertificate) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionViewModel.kt:77]: `items.value?.isNotEmpty()` ? and no need for `hasValidCertificate`

Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionViewModel.kt:77]: I'm not checking `items` but `certificates`

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionViewModel.kt:77]: Ah Ok, please make hasValidCertificate `private` ? it is only used in the VM

Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionViewModel.kt:77]: updated

**Review by mtwalli — APPROVED** [review]

**Review by BMItr — APPROVED** [review]

**Pull request merged**